distributed/core.py

```diff
@@ -1007,28 +1014,39 @@ async def connect(self, addr, timeout=None):
 
         self._n_connecting += 1
         await self.semaphore.acquire()
-
+        fut = None
         try:
-            comm = await connect(
-                addr,
-                timeout=timeout or self.timeout,
-                deserialize=self.deserialize,
-                **self.connection_args,
+            if self.status != Status.running:
+                raise CommClosedError("ConnectionPool closed already")
```

Comment:
Maybe this can't happen, but what if it's starting?  
```suggestion
                raise CommClosedError(f"ConnectionPool not running.  Status: {self.status}")
```

Reply:
I _think_ this can never happen. the start initializes the semaphore. without semaphore this entire thing would break. Still, your exception message is nicer and more future proof.